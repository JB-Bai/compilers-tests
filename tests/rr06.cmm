float Sw5JSQr7(float VTZG6M){
	struct TLOj38 Eqiod;
	struct{
		struct x91tbai{
		}
		DHPqEmtR_B;
		float tA4lEtq[888][3550][7750][7472];
	}
	DSDbpJe6W[37541], jXZeArCLkr7 = 437.9685 / RuKJoXo;
	{
		if(Sf0xxcps(jKK_aKN) <= t4tiXX2w0y(44) + SJpoG0XVD / ! ! 01 / ! 95436 - 44.7231[twirEvXzkc[! rTX9LwK - W14wOed5()][106]] || 0x8c[265.14]) 85.86E5 &&(Rwvx3SQ() && 144 == zyTixht + 761.055);
		else while(0xeF7) if(5716.09) if(agazlDkk) xAdHDxlb;
		while((Lnrs2kF)) mhZ;
		{
			1500;
			if(yk2tYopUwl) DA9Jt && ! qavp3.S7SbUYVu;
			else if(3632.112 || wkD8G(607.460)) if(UJEBVkNB[030117]) ! clNGC1_ - 0xCcC[6.20.qfj8rdxLP * 60.26 * SJUDNJQo7() / ! 5.7368 / 060354 > 6931.56 - ! 3772 && pogyypl / 9.926e+7 * 92.19 + 9.9E-1 && 441.81] = ! 6834.1 || 010.05E+3;
			else return FsUlIJe;
			96.379.glK8KMRbX3vC[! 1890 * 0xa90f - 0xad28 = ueiYh3z];
			! 0x95A1;
		}
	}
	(0x2E7D || fcng_ + 28.3864e3 && WusT0QB && uDLZ3i = 5.9 && ! mlUtjz || 802.715e1 / 2249.07[DbQdAnE]);
}
struct bsQ7KWT JZu6WwkS(){
	struct{
		struct D4xvhzsJU{
		}
		rMUvxuU = 131;
		float lU3JyM = 84.06[057.63] / lLHg(), m9s3xF7l;
	}
	xnHdj8nZ0i, l37wD34c = 3603.84E0;
	float k_QieL4G;
	if(! 0x3264 = 29176 + 64.676e1) return 3749.082E4;
	if(hwiwM()) lQSGC;
	else 0x12;
	return wBVQ38fzBQ * 0xC;
	sKYp0_dfSb();
}
float lIBoAhJi(){
	00.4e+4;
	17.858E+6 *(o5jK74aI / 8.38[99[q5Lbr9z() = ! _O6VRlIhrK *(6.343e1 * 0274 * _S9lm1M() / - 0x5DD -(3745.985e+2 - 0x8ED) != hkYc() * d6EFuhcY[89.368] * jXxGoQd || YjE4fUgwSR / ffJmBuKzP(7.7577e2 && 69.2634E6 + 32.939E+8 && KmKTwfI() * 0399.237e1 * 619.756.FPcj0_Ne4g * hzLzxXQ || 0x6Ed8 && - 05734 * 4552.07 || 0612) && 0xaB2 * 04342 / 6.467) == GasvsN(0042)[0xF1D] * qkZZhi]]) / 7354.92e0;
	(wTMLNNuKzL(OhdIfC));
}

