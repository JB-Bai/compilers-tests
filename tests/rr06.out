Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      FunDec (1)
        ID: Sw5JSQr7
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: float
            VarDec (1)
              ID: VTZG6M
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              StructSpecifier (2)
                STRUCT
                Tag (2)
                  ID: TLOj38
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: Eqiod
            SEMI
          DefList (3)
            Def (3)
              Specifier (3)
                StructSpecifier (3)
                  STRUCT
                  LC
                  DefList (4)
                    Def (4)
                      Specifier (4)
                        StructSpecifier (4)
                          STRUCT
                          OptTag (4)
                            ID: x91tbai
                          LC
                          RC
                      DecList (6)
                        Dec (6)
                          VarDec (6)
                            ID: DHPqEmtR_B
                      SEMI
                    DefList (7)
                      Def (7)
                        Specifier (7)
                          TYPE: float
                        DecList (7)
                          Dec (7)
                            VarDec (7)
                              VarDec (7)
                                VarDec (7)
                                  VarDec (7)
                                    VarDec (7)
                                      ID: tA4lEtq
                                    LB
                                    INT: 888
                                    RB
                                  LB
                                  INT: 3550
                                  RB
                                LB
                                INT: 7750
                                RB
                              LB
                              INT: 7472
                              RB
                        SEMI
                  RC
              DecList (9)
                Dec (9)
                  VarDec (9)
                    VarDec (9)
                      ID: DSDbpJe6W
                    LB
                    INT: 37541
                    RB
                COMMA
                DecList (9)
                  Dec (9)
                    VarDec (9)
                      ID: jXZeArCLkr7
                    ASSIGNOP
                    Exp (9)
                      Exp (9)
                        FLOAT: 437.968506
                      DIV
                      Exp (9)
                        ID: RuKJoXo
              SEMI
        StmtList (10)
          Stmt (10)
            CompSt (10)
              LC
              StmtList (11)
                Stmt (11)
                  IF
                  LP
                  Exp (11)
                    Exp (11)
                      Exp (11)
                        ID: Sf0xxcps
                        LP
                        Args (11)
                          Exp (11)
                            ID: jKK_aKN
                        RP
                      RELOP
                      Exp (11)
                        Exp (11)
                          Exp (11)
                            ID: t4tiXX2w0y
                            LP
                            Args (11)
                              Exp (11)
                                INT: 44
                            RP
                          PLUS
                          Exp (11)
                            Exp (11)
                              Exp (11)
                                ID: SJpoG0XVD
                              DIV
                              Exp (11)
                                NOT
                                Exp (11)
                                  NOT
                                  Exp (11)
                                    INT: 1
                            DIV
                            Exp (11)
                              NOT
                              Exp (11)
                                INT: 95436
                        MINUS
                        Exp (11)
                          Exp (11)
                            FLOAT: 44.723099
                          LB
                          Exp (11)
                            Exp (11)
                              Exp (11)
                                ID: twirEvXzkc
                              LB
                              Exp (11)
                                Exp (11)
                                  NOT
                                  Exp (11)
                                    ID: rTX9LwK
                                MINUS
                                Exp (11)
                                  ID: W14wOed5
                                  LP
                                  RP
                              RB
                            LB
                            Exp (11)
                              INT: 106
                            RB
                          RB
                    OR
                    Exp (11)
                      Exp (11)
                        INT: 140
                      LB
                      Exp (11)
                        FLOAT: 265.140015
                      RB
                  RP
                  Stmt (11)
                    Exp (11)
                      Exp (11)
                        FLOAT: 8586000.000000
                      AND
                      Exp (11)
                        LP
                        Exp (11)
                          Exp (11)
                            ID: Rwvx3SQ
                            LP
                            RP
                          AND
                          Exp (11)
                            Exp (11)
                              INT: 144
                            RELOP
                            Exp (11)
                              Exp (11)
                                ID: zyTixht
                              PLUS
                              Exp (11)
                                FLOAT: 761.054993
                        RP
                    SEMI
                  ELSE
                  Stmt (12)
                    WHILE
                    LP
                    Exp (12)
                      INT: 3831
                    RP
                    Stmt (12)
                      IF
                      LP
                      Exp (12)
                        FLOAT: 5716.089844
                      RP
                      Stmt (12)
                        IF
                        LP
                        Exp (12)
                          ID: agazlDkk
                        RP
                        Stmt (12)
                          Exp (12)
                            ID: xAdHDxlb
                          SEMI
                StmtList (13)
                  Stmt (13)
                    WHILE
                    LP
                    Exp (13)
                      LP
                      Exp (13)
                        ID: Lnrs2kF
                      RP
                    RP
                    Stmt (13)
                      Exp (13)
                        ID: mhZ
                      SEMI
                  StmtList (14)
                    Stmt (14)
                      CompSt (14)
                        LC
                        StmtList (15)
                          Stmt (15)
                            Exp (15)
                              INT: 1500
                            SEMI
                          StmtList (16)
                            Stmt (16)
                              IF
                              LP
                              Exp (16)
                                ID: yk2tYopUwl
                              RP
                              Stmt (16)
                                Exp (16)
                                  Exp (16)
                                    ID: DA9Jt
                                  AND
                                  Exp (16)
                                    NOT
                                    Exp (16)
                                      Exp (16)
                                        ID: qavp3
                                      DOT
                                      ID: S7SbUYVu
                                SEMI
                              ELSE
                              Stmt (17)
                                IF
                                LP
                                Exp (17)
                                  Exp (17)
                                    FLOAT: 3632.112061
                                  OR
                                  Exp (17)
                                    ID: wkD8G
                                    LP
                                    Args (17)
                                      Exp (17)
                                        FLOAT: 607.460022
                                    RP
                                RP
                                Stmt (17)
                                  IF
                                  LP
                                  Exp (17)
                                    Exp (17)
                                      ID: UJEBVkNB
                                    LB
                                    Exp (17)
                                      INT: 12367
                                    RB
                                  RP
                                  Stmt (17)
                                    Exp (17)
                                      Exp (17)
                                        Exp (17)
                                          NOT
                                          Exp (17)
                                            ID: clNGC1_
                                        MINUS
                                        Exp (17)
                                          Exp (17)
                                            INT: 3276
                                          LB
                                          Exp (17)
                                            Exp (17)
                                              Exp (17)
                                                Exp (17)
                                                  Exp (17)
                                                    Exp (17)
                                                      Exp (17)
                                                        Exp (17)
                                                          Exp (17)
                                                            FLOAT: 6.200000
                                                          DOT
                                                          ID: qfj8rdxLP
                                                        STAR
                                                        Exp (17)
                                                          FLOAT: 60.259998
                                                      STAR
                                                      Exp (17)
                                                        ID: SJUDNJQo7
                                                        LP
                                                        RP
                                                    DIV
                                                    Exp (17)
                                                      NOT
                                                      Exp (17)
                                                        FLOAT: 5.736800
                                                  DIV
                                                  Exp (17)
                                                    INT: 24812
                                                RELOP
                                                Exp (17)
                                                  Exp (17)
                                                    FLOAT: 6931.560059
                                                  MINUS
                                                  Exp (17)
                                                    NOT
                                                    Exp (17)
                                                      INT: 3772
                                              AND
                                              Exp (17)
                                                Exp (17)
                                                  Exp (17)
                                                    Exp (17)
                                                      ID: pogyypl
                                                    DIV
                                                    Exp (17)
                                                      FLOAT: 99260000.000000
                                                  STAR
                                                  Exp (17)
                                                    FLOAT: 92.190002
                                                PLUS
                                                Exp (17)
                                                  FLOAT: 0.990000
                                            AND
                                            Exp (17)
                                              FLOAT: 441.809998
                                          RB
                                      ASSIGNOP
                                      Exp (17)
                                        Exp (17)
                                          NOT
                                          Exp (17)
                                            FLOAT: 6834.100098
                                        OR
                                        Exp (17)
                                          FLOAT: 10050.000000
                                    SEMI
                                  ELSE
                                  Stmt (18)
                                    RETURN
                                    Exp (18)
                                      ID: FsUlIJe
                                    SEMI
                            StmtList (19)
                              Stmt (19)
                                Exp (19)
                                  Exp (19)
                                    Exp (19)
                                      FLOAT: 96.378998
                                    DOT
                                    ID: glK8KMRbX3vC
                                  LB
                                  Exp (19)
                                    Exp (19)
                                      Exp (19)
                                        Exp (19)
                                          NOT
                                          Exp (19)
                                            INT: 1890
                                        STAR
                                        Exp (19)
                                          INT: 43279
                                      MINUS
                                      Exp (19)
                                        INT: 44328
                                    ASSIGNOP
                                    Exp (19)
                                      ID: ueiYh3z
                                  RB
                                SEMI
                              StmtList (20)
                                Stmt (20)
                                  Exp (20)
                                    NOT
                                    Exp (20)
                                      INT: 38305
                                  SEMI
                        RC
              RC
          StmtList (23)
            Stmt (23)
              Exp (23)
                LP
                Exp (23)
                  Exp (23)
                    Exp (23)
                      INT: 11901
                    OR
                    Exp (23)
                      Exp (23)
                        Exp (23)
                          Exp (23)
                            ID: fcng_
                          PLUS
                          Exp (23)
                            FLOAT: 28386.400391
                        AND
                        Exp (23)
                          ID: WusT0QB
                      AND
                      Exp (23)
                        ID: uDLZ3i
                  ASSIGNOP
                  Exp (23)
                    Exp (23)
                      Exp (23)
                        FLOAT: 5.900000
                      AND
                      Exp (23)
                        NOT
                        Exp (23)
                          ID: mlUtjz
                    OR
                    Exp (23)
                      Exp (23)
                        FLOAT: 8027.149902
                      DIV
                      Exp (23)
                        Exp (23)
                          FLOAT: 2249.070068
                        LB
                        Exp (23)
                          ID: DbQdAnE
                        RB
                RP
              SEMI
        RC
    ExtDefList (25)
      ExtDef (25)
        Specifier (25)
          StructSpecifier (25)
            STRUCT
            Tag (25)
              ID: bsQ7KWT
        FunDec (25)
          ID: JZu6WwkS
          LP
          RP
        CompSt (25)
          LC
          DefList (26)
            Def (26)
              Specifier (26)
                StructSpecifier (26)
                  STRUCT
                  LC
                  DefList (27)
                    Def (27)
                      Specifier (27)
                        StructSpecifier (27)
                          STRUCT
                          OptTag (27)
                            ID: D4xvhzsJU
                          LC
                          RC
                      DecList (29)
                        Dec (29)
                          VarDec (29)
                            ID: rMUvxuU
                          ASSIGNOP
                          Exp (29)
                            INT: 131
                      SEMI
                    DefList (30)
                      Def (30)
                        Specifier (30)
                          TYPE: float
                        DecList (30)
                          Dec (30)
                            VarDec (30)
                              ID: lU3JyM
                            ASSIGNOP
                            Exp (30)
                              Exp (30)
                                Exp (30)
                                  FLOAT: 84.059998
                                LB
                                Exp (30)
                                  FLOAT: 57.630001
                                RB
                              DIV
                              Exp (30)
                                ID: lLHg
                                LP
                                RP
                          COMMA
                          DecList (30)
                            Dec (30)
                              VarDec (30)
                                ID: m9s3xF7l
                        SEMI
                  RC
              DecList (32)
                Dec (32)
                  VarDec (32)
                    ID: xnHdj8nZ0i
                COMMA
                DecList (32)
                  Dec (32)
                    VarDec (32)
                      ID: l37wD34c
                    ASSIGNOP
                    Exp (32)
                      FLOAT: 3603.840088
              SEMI
            DefList (33)
              Def (33)
                Specifier (33)
                  TYPE: float
                DecList (33)
                  Dec (33)
                    VarDec (33)
                      ID: k_QieL4G
                SEMI
          StmtList (34)
            Stmt (34)
              IF
              LP
              Exp (34)
                Exp (34)
                  NOT
                  Exp (34)
                    INT: 12900
                ASSIGNOP
                Exp (34)
                  Exp (34)
                    INT: 29176
                  PLUS
                  Exp (34)
                    FLOAT: 646.760010
              RP
              Stmt (34)
                RETURN
                Exp (34)
                  FLOAT: 37490820.000000
                SEMI
            StmtList (35)
              Stmt (35)
                IF
                LP
                Exp (35)
                  ID: hwiwM
                  LP
                  RP
                RP
                Stmt (35)
                  Exp (35)
                    ID: lQSGC
                  SEMI
                ELSE
                Stmt (36)
                  Exp (36)
                    INT: 18
                  SEMI
              StmtList (37)
                Stmt (37)
                  RETURN
                  Exp (37)
                    Exp (37)
                      ID: wBVQ38fzBQ
                    STAR
                    Exp (37)
                      INT: 12
                  SEMI
                StmtList (38)
                  Stmt (38)
                    Exp (38)
                      ID: sKYp0_dfSb
                      LP
                      RP
                    SEMI
          RC
      ExtDefList (40)
        ExtDef (40)
          Specifier (40)
            TYPE: float
          FunDec (40)
            ID: lIBoAhJi
            LP
            RP
          CompSt (40)
            LC
            StmtList (41)
              Stmt (41)
                Exp (41)
                  FLOAT: 4000.000000
                SEMI
              StmtList (42)
                Stmt (42)
                  Exp (42)
                    Exp (42)
                      Exp (42)
                        FLOAT: 17858000.000000
                      STAR
                      Exp (42)
                        LP
                        Exp (42)
                          Exp (42)
                            ID: o5jK74aI
                          DIV
                          Exp (42)
                            Exp (42)
                              FLOAT: 8.380000
                            LB
                            Exp (42)
                              Exp (42)
                                INT: 99
                              LB
                              Exp (42)
                                Exp (42)
                                  ID: q5Lbr9z
                                  LP
                                  RP
                                ASSIGNOP
                                Exp (42)
                                  Exp (42)
                                    Exp (42)
                                      NOT
                                      Exp (42)
                                        ID: _O6VRlIhrK
                                    STAR
                                    Exp (42)
                                      LP
                                      Exp (42)
                                        Exp (42)
                                          Exp (42)
                                            Exp (42)
                                              Exp (42)
                                                Exp (42)
                                                  Exp (42)
                                                    FLOAT: 63.430000
                                                  STAR
                                                  Exp (42)
                                                    INT: 188
                                                STAR
                                                Exp (42)
                                                  ID: _S9lm1M
                                                  LP
                                                  RP
                                              DIV
                                              Exp (42)
                                                MINUS
                                                Exp (42)
                                                  INT: 1501
                                            MINUS
                                            Exp (42)
                                              LP
                                              Exp (42)
                                                Exp (42)
                                                  FLOAT: 374598.500000
                                                MINUS
                                                Exp (42)
                                                  INT: 2285
                                              RP
                                          RELOP
                                          Exp (42)
                                            Exp (42)
                                              Exp (42)
                                                ID: hkYc
                                                LP
                                                RP
                                              STAR
                                              Exp (42)
                                                Exp (42)
                                                  ID: d6EFuhcY
                                                LB
                                                Exp (42)
                                                  FLOAT: 89.367996
                                                RB
                                            STAR
                                            Exp (42)
                                              ID: jXxGoQd
                                        OR
                                        Exp (42)
                                          Exp (42)
                                            Exp (42)
                                              ID: YjE4fUgwSR
                                            DIV
                                            Exp (42)
                                              ID: ffJmBuKzP
                                              LP
                                              Args (42)
                                                Exp (42)
                                                  Exp (42)
                                                    Exp (42)
                                                      Exp (42)
                                                        Exp (42)
                                                          FLOAT: 775.770020
                                                        AND
                                                        Exp (42)
                                                          Exp (42)
                                                            FLOAT: 69263400.000000
                                                          PLUS
                                                          Exp (42)
                                                            FLOAT: 3293900032.000000
                                                      AND
                                                      Exp (42)
                                                        Exp (42)
                                                          Exp (42)
                                                            Exp (42)
                                                              ID: KmKTwfI
                                                              LP
                                                              RP
                                                            STAR
                                                            Exp (42)
                                                              FLOAT: 3992.370117
                                                          STAR
                                                          Exp (42)
                                                            Exp (42)
                                                              FLOAT: 619.755981
                                                            DOT
                                                            ID: FPcj0_Ne4g
                                                        STAR
                                                        Exp (42)
                                                          ID: hzLzxXQ
                                                    OR
                                                    Exp (42)
                                                      Exp (42)
                                                        INT: 28376
                                                      AND
                                                      Exp (42)
                                                        Exp (42)
                                                          MINUS
                                                          Exp (42)
                                                            INT: 3036
                                                        STAR
                                                        Exp (42)
                                                          FLOAT: 4552.069824
                                                  OR
                                                  Exp (42)
                                                    INT: 394
                                              RP
                                          AND
                                          Exp (42)
                                            Exp (42)
                                              Exp (42)
                                                INT: 2738
                                              STAR
                                              Exp (42)
                                                INT: 2274
                                            DIV
                                            Exp (42)
                                              FLOAT: 6.467000
                                      RP
                                  RELOP
                                  Exp (42)
                                    Exp (42)
                                      Exp (42)
                                        ID: GasvsN
                                        LP
                                        Args (42)
                                          Exp (42)
                                            INT: 34
                                        RP
                                      LB
                                      Exp (42)
                                        INT: 3869
                                      RB
                                    STAR
                                    Exp (42)
                                      ID: qkZZhi
                              RB
                            RB
                        RP
                    DIV
                    Exp (42)
                      FLOAT: 7354.919922
                  SEMI
                StmtList (43)
                  Stmt (43)
                    Exp (43)
                      LP
                      Exp (43)
                        ID: wTMLNNuKzL
                        LP
                        Args (43)
                          Exp (43)
                            ID: OhdIfC
                        RP
                      RP
                    SEMI
            RC
